paramiko/transport.py

```diff
@@ -1038,13 +1043,17 @@ def global_request(self, kind, data=None, wait=True):
         self._send_user_message(m)
         if not wait:
             return None
-        while True:
+        # sets default timeout to 5 in case ``wait`` == True
+        timeout = 5 if isinstance(wait, bool) else wait
```

Comment:
This is a bit awkward - there's no elegant value to mean "forever" (which might be useless), and 5 is written in two places.

Here's another idea: let `True` mean "forever", and change all the places in paramiko.transport which call `global_request(... wait=True)` to just not specify wait, allowing the default to be used. Then, perhaps in a later PR, the appropriate timeout configured in the Transport could be specified when `global_request()` is called.

Reply:
Also, the other timeouts in Transport have higher default values, 15 or 30 seconds.